sdk/tests/wallet/output_preparation.rs

```diff
@@ -557,3 +559,192 @@ async fn prepare_nft_output_features_update() -> Result<()> {
+                assets: None,
+                features: None,
+                unlocks: None,
+                storage_deposit: Some(StorageDeposit {
+                    return_strategy: Some(ReturnStrategy::Gift),
+                    use_excess_if_low: Some(true),
+                }),
+            },
+            None,
+        )
+        .await?;
+
+    // Check if the output has enough amount to cover the storage deposit
+    output.verify_storage_deposit(rent_structure, token_supply)?;
+    // The left over 21299 is too small to keep, so we donate it
+    assert_eq!(output.amount(), balance.base_coin().available());
+    // storage deposit gifted, only address unlock condition
+    assert_eq!(output.unlock_conditions().unwrap().len(), 1);
+
+    let result = account
+        .prepare_output(
+            OutputParams {
+                recipient_address: *address,
+                amount: minimum_required_storage_deposit - 1, // Leave less than min. deposit
+                assets: None,
+                features: None,
+                unlocks: None,
+                storage_deposit: Some(StorageDeposit {
+                    return_strategy: Some(ReturnStrategy::Return),
+                    use_excess_if_low: Some(true),
+                }),
+            },
+            None,
+        )
+        .await;
+    matches!(result, Err(iota_sdk::wallet::Error::InsufficientFunds{available, required}) if available == balance.base_coin().available() && required == balance.base_coin().available()+minimum_required_storage_deposit-1);
+
+    let output = account
+        .prepare_output(
+            OutputParams {
+                recipient_address: *address,
+                amount: 100, // leave more than min. deposit
+                assets: None,
+                features: None,
+                unlocks: None,
+                storage_deposit: Some(StorageDeposit {
+                    return_strategy: Some(ReturnStrategy::Gift),
+                    use_excess_if_low: Some(true),
+                }),
+            },
+            None,
+        )
+        .await?;
+
+    // Check if the output has enough amount to cover the storage deposit
+    output.verify_storage_deposit(rent_structure, token_supply)?;
+    // We use excess if leftover is too small, so amount == all available balance
+    assert_eq!(output.amount(), 63900);
+    // storage deposit gifted, only address unlock condition
+    assert_eq!(output.unlock_conditions().unwrap().len(), 1);
+
+    let output = account
+        .prepare_output(
+            OutputParams {
+                recipient_address: *address,
+                amount: 100, // leave more than min. deposit
+                assets: None,
+                features: None,
+                unlocks: None,
+                storage_deposit: Some(StorageDeposit {
+                    return_strategy: Some(ReturnStrategy::Return),
+                    use_excess_if_low: Some(true),
```

Comment:
Should we have a test with false?

Reply:
false is used in the general output_preparation tests as we set `use_excess_if_low: None`

Reply:
Yeah false is default